azurerm/mssql-database/variables.tf

```diff
@@ -1,63 +1,100 @@
+  }
 }
 
 variable "short_term_retention_policy_in_days" {
   # https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_database#short_term_retention_policy
-  description = "(Optional) Point In Time Restore configuration. Value has to be between 7 and 35."
+  description = "(Optional) Point In Time Restore configuration. Value has to be between 1 and 35."
   default     = "7"
+
+  validation {
+    condition     = var.short_term_retention_policy_in_days >= 1 && var.short_term_retention_policy_in_days <= 35
+    error_message = "Number of retention days has to be between 1 and 35."
+  }
 }
 
 variable "auto_pause_delay_in_minutes" {
-  description = "(Optional) Time in minutes after which database is automatically paused. A value of -1 means that automatic pause is disabled. This property is only settable for General Purpose Serverless databases. Value should be in increments of 10, e.g. `10`, `20`, `30`, etc."
+  description = "(Optional) Time in minutes after which database is automatically paused. A value of -1 means that automatic pause is disabled. This property is only settable for General Purpose Serverless databases. Value should be in increments of 10, e.g. `10`, `20`, `30`, etc. Maximum is 10080 (seven days)."
   default     = 60
   type        = number
+
+  validation {
+    condition     = var.auto_pause_delay_in_minutes == -1 || (var.auto_pause_delay_in_minutes >= 15 && var.auto_pause_delay_in_minutes <= 10080)
+    error_message = "Invalid Auto Pause Delay value. Value should be either -1 or between 15 and 10080 (seven days)."
+  }
 }
 
 variable "collation" {
+  //TODO: Sjekk om det er nødvendig å lage validation for collation
   description = "(Optional) Specifies the collation of the database. Changing this forces a new resource to be created."
   default     = "SQL_LATIN1_GENERAL_CP1_CI_AS"
 }
 
 variable "max_size_gb" {
   description = "(Optional) The max size of the database in gigabytes."
   default     = null
+
+  validation {
+    condition     = var.max_size_gb == null || var.max_size_gb >= 1
+    error_message = "Invalid Max Size (GB) value. Value should be greater than or equal to 1."
+  }
 }
 
 variable "min_capacity" {
-  description = "(Optional) Minimal capacity that database will always have allocated, if not paused. This property is only settable for General Purpose Serverless databases."
+  description = "(Optional) Minimal capacity (vCore) that database will always have allocated, if not paused. This property is only settable for General Purpose Serverless databases."
   default     = 0.5
+
+  validation {
+    condition     = var.min_capacity >= 0.5 && var.min_capacity <= 128
+    error_message = "Invalid Min Capacity (vCore) value. Value should be between 0.5 and 128."
+  }
 }
 
 variable "weekly_backup_retention_period" {
```

Comment:
Denne kan vi skrive en validation for, men det blir fort en hissig regex. Fant ingen pen måte å gjøre det på